homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,244 @@
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, temperature):
+        """Set new target temperature."""
+        # pylint: disable=too-many-branches
+        if self._zxt_120 == 1:
+            # ZXT-120 does not support get setpoint
+            self._target_temperature = temperature
+            if self._current_operation == 'Heat':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 1:
+                        # ZXT-120 only responds to whole int
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Cool':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 2:
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Dry Air':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 8:
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Auto Changeover':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 10:
+                        value.data = int(round(temperature, 0))
+        return
+
+    def set_fan_mode(self, current_fan_mode):
+        """Set new target fan mode."""
+        # set point
+        if self._zxt_120 == 1:
+            for value in self._node.get_values(class_id=0x44).values():
+                if value.command_class == 68 and value.index == 0:
+                    value.data = bytes(current_fan_mode, 'utf-8')
+        return
+
+    def set_operation(self, current_operation):
+        """Set new target operation mode."""
+        if self._zxt_120 == 1:
+            for value in self._node.get_values(class_id=0x40).values():
+                if value.command_class == 64 and value.index == 0:
+                    value.data = bytes(current_operation, 'utf-8')
+        return
+
+    def set_swing(self, swing_mode):
+        """Set new target swing mode."""
+        if self._zxt_120 == 1:
+            for value in self._node.get_values(class_id=0x70).values():
+                if value.command_class == 112 and value.index == 33:
+                    value.data = int(swing_mode)
+        return
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self._zxt_120 == 1:
```

Comment:
Let's just start with 19. I know my ac is 19 too

On Tue, Apr 26, 2016, 13:21 John Arild Berentsen <notifications@github.com>
wrote:

> In homeassistant/components/hvac/zwave.py
> <https://github.com/home-assistant/home-assistant/pull/1912#discussion_r61068592>
> :
>
> > +                if value.command_class == 64 and value.index == 0:
> > +                    value.data = bytes(current_operation, 'utf-8')
> > +        return
> > +
> > +    def set_swing(self, swing_mode):
> > +        """Set new target swing mode."""
> > +        if self._zxt_120 == 1:
> > +            for value in self._node.get_values(class_id=0x70).values():
> > +                if value.command_class == 112 and value.index == 33:
> > +                    value.data = int(swing_mode)
> > +        return
> > +
> > +    @property
> > +    def min_temp(self):
> > +        """Return the minimum temperature."""
> > +        if self._zxt_120 == 1:
>
> I do not know how low setpoint are beeing used for other hvacs. The
> zxt-120 is only 19Celsius, but other might be 15. Do not want to block
> tmeps before I know for sure.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/1912/files/403c27395c6cc511c04f1b241a05b95da265f2b0#r61068592>
>